community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/TxStateTransactionDataSnapshot.java

```diff
@@ -170,26 +171,26 @@ private void takeSnapshot()
         {
             for ( Long nodeId : state.addedAndRemovedNodes().getRemoved() )
             {
-                try ( NodeCursor node = storeStatement.acquireSingleNodeCursor( nodeId ) )
+                try ( Cursor<NodeItem> node = storeStatement.acquireSingleNodeCursor( nodeId ) )
                 {
                     if ( node.next() )
                     {
-                        try ( PropertyCursor properties = node.properties() )
+                        try ( Cursor<PropertyItem> properties = node.get().properties() )
                         {
                             while ( properties.next() )
                             {
                                 removedNodeProperties.add( new NodePropertyEntryView( nodeId,
-                                        store.propertyKeyGetName( properties.propertyKeyId() ), null,
-                                        properties.value() ) );
+                                        store.propertyKeyGetName( properties.get().propertyKeyId() ), null,
```

Comment:
These two `properties.get()` could be optimized into one, but the only cost of doing two would be to make the null check in the property Cursor two times.